attribute_set_completeness/models/attribute_set_owner_mixin.py

```diff
@@ -29,24 +31,42 @@ class AttributeSetOwnerMixin(models.AbstractModel):
 
     @api.multi
     def _compute_completion_rate(self):
-        for record in self:
-            attribute_set_id = record.attribute_set_id
+        for rec in self:
+            attribute_set_id = rec.attribute_set_id
             completion_config = attribute_set_id.attribute_set_completeness_ids
             if completion_config:
                 completion_rate = 0.0
                 not_complete_crit = self.env["attribute.set.completeness"].browse()
                 for criteria in completion_config:
                     field_name = criteria.field_id.name
-                    if record[field_name]:
+                    if rec[field_name]:
                         completion_rate += criteria.completion_rate
                     else:
                         not_complete_crit |= criteria
-                record.completion_rate = completion_rate
-                record.attribute_set_not_completed_ids = not_complete_crit
+                rec.completion_rate = completion_rate
+                rec.attribute_set_not_completed_ids = not_complete_crit
                 if completion_rate < 100:
-                    record.completion_state = "not_complete"
+                    rec.completion_state = "not_complete"
                 else:
-                    record.completion_state = "complete"
+                    rec.completion_state = "complete"
             else:
-                record.completion_rate = 0
-                record.completion_state = "not_complete"
+                rec.completion_rate = 0
+                rec.completion_state = "not_complete"
+
+    @api.multi
+    def _compute_is_completion_state_visible(self):
```

Comment:
```suggestion
    @api.depends("attribute_set_id", "attribute_set_id.attribute_set_completeness_ids")
    def _compute_is_completion_state_visible(self):
```

Reply:
Hi @lmignon , in fact I didn't know if the `@api.depends` were needed here as they are not used in `_compute_completion_rate`.
I understand now that if the compute method is taken care by the `AbstractComponent` model (with this quite rare `@skip_if` decorator) we need to add the `@api.multi` decorator, but if our compute method is not in this component model, we need to add the `@api.depends` as usual (and the fact that's it's in a mixin doesn't matter)... Is it right ?

Anyway thank you very much for all you advises !